src/stream-tcp-reassemble.c

```diff
@@ -1049,14 +1050,21 @@ static int ReassembleUpdateAppLayer (ThreadVars *tv,
                 return 0;
 
             continue;
-        } else if (mydata == NULL || mydata_len == 0) {
+        } else if (flags & STREAM_DEPTH) {
+            // we're just called once with this flag, so make sure we pass it on
+            if (mydata == NULL && mydata_len > 0) {
+                mydata_len = 0;
+            }
+        } else if (mydata == NULL || (mydata_len == 0 && ((flags & STREAM_EOF) == 0))) {
```

Comment:
the `STREAM_EOF` check here comes from another commit (b6fed6d431bb09b573555b2560eb805a59cd91ec). Why was that added here?